### DISCUSSION TITLE
xDrip Sync – Master does not receive calibrations from Followers

### DISCUSSION THREAD

**Discussion by AliQ101** [NightscoutFoundation/xDrip#3699] [Bug reports]
YMaster: Samsung (Android 14, One UI 6.1, Knox 3.10), Followers: HTC, Samsung devices (Android 9/14). Master’s source is Libre2/OOP. Master and Followers devices are running the latest stable xDrip 17/9/24 (dfcbe80-2024.09.17). xDrip Cloud is enabled on all devices.

Currently, the Follower’s devices are able send treatments (carb, insulin) and notes to Master or change/delete treatments/notes entered by Master or change/delete treatments entered by the other Follower for Master. That’s work fine. However, I have recognized that Master does not receive calibrations from Follower.

Test:
Step 1) Follower entered calibration of 5.8 at 09:10, pop-up message „Calibration sent to master for processing“ on Follower device. No calibration appeared on Master.
Step2) Follower entered calibration of 5.9 at 09:12, the same as above.
See attached logs from Master for time 09:10 and 09:12.
![IMAG9815~2](https://github.com/user-attachments/assets/adc2067d-c293-4ce4-b900-16ebdec50e01)

Step3) Now before entering new calibration from Follower I have opened xDrip application on Master device’s screen. Then Follower entered calibration of 6.0 at 09:16, pop-up message appeared on Follower, and after while calibration appeared on Master. See attached logs from Master for time 09:16 and accepted calibration on Master.
![IMAG9816~2](https://github.com/user-attachments/assets/df86942e-fcff-463b-9375-18a6ec2b58e7)
![IMAG9817~2](https://github.com/user-attachments/assets/8db80a6b-266e-44fe-a950-4fb8c43227da)
![IMAG9818~2](https://github.com/user-attachments/assets/03d201bc-5f30-4952-a5b0-aab10b66d80c)

The above happens no matter xDrip Cloud option is on/off. About 2-3 months ago we have changed Master phone (Samsung/Android 9) to current one, xDrip settings remained the same. With previous model phone all worked fine including calibrations. Not sure if this issue relates to xDrip or phone as other actions by Followers are accepted. Any ideas?

**Comment by Navid200** [comment]
Can you post screenshots of the "Followers" status page from the master and followers?

**Reply by AliQ101** [reply]
Master and two Followers (one was used for test), third Follower is out of reach now:
![IMAG9822~2](https://github.com/user-attachments/assets/0d2c0916-e805-4e8f-91df-06d21d165d6f)
![IMAG9821](https://github.com/user-attachments/assets/9abf29d8-f8df-4d33-90bf-ba5a342618b4)
![Screenshot_20241005-213506](https://github.com/user-attachments/assets/02e03c69-7a94-41b5-b4e7-c0ed3d86fcbd)

**Comment by Navid200** [comment]
I have read your description of the the tests.

So, if you have xDrip open on the master, calibrations will go from the follower to the master.  But, if xDrip is closed on the master. they will not.
Is that it?

**Reply by AliQ101** [reply]
Yes, exactly.

**Reply by Navid200** [reply]
But, the logs you have posted show that calibrations at 9:10 and 9:12 have also been accepted.
Why do you think they have not been accepted?  Because of the calibration data table?

**Reply by AliQ101** [reply]
Where from the logs at 09:10 and 09:12 can be seen that calibrations 5.8 and 5.9 were accepted/used? I can see only cgmactivity but not followed by add calibration as at 09:16 for 6.0. 
![IMAG9817~3](https://github.com/user-attachments/assets/7899d250-91b0-48c0-a39a-e5b67bd9a8ff)

Before calibrations by 5.8, 5.9 and 6.0, xdrip value was 5.4-5.5,  they were all in range to be accepted. At 9:10 and 9:12 there were not calibration dot shown on xdrip's main screen on Master/Follower, no show in calibration table on Master, no move in glycemia curve on Master.

**Reply by Navid200** [reply]
OK,  I see what you mean now.

I don't have Libre.  I use Dexcom G7, xDrip 2024.09.27.
I just set up an xDrip sync follower, 2024.09.19.  I closed xDrip on the master, and locked the screen on the master.
Then, I submitted a calibration from the follower.  It was received by the master and was entered in to the transmitter queue to be submitted to the transmitter.  

So, it seems that it works fine for me.  I don't know why the hardware data source would have an impact on this.  

@jamorham Any idea?
